repair/repair.cc

```diff
@@ -695,14 +696,18 @@ void repair::shard_repair_task_impl::check_in_abort_or_shutdown() {
 
 repair_neighbors repair::shard_repair_task_impl::get_repair_neighbors(const dht::token_range& range) {
     return neighbors.empty() ?
-        repair_neighbors(get_neighbors(gossiper, *erm, _status.keyspace, range, data_centers, hosts, ignore_nodes, _small_table_optimization)) :
+        repair_neighbors(get_neighbors(gossiper, *get_erm(), _status.keyspace, range, data_centers, hosts, ignore_nodes, _small_table_optimization)) :
         neighbors[range];
 }
 
 size_t repair::shard_repair_task_impl::ranges_size() const noexcept {
     return ranges.size() * table_ids.size();
 }
 
+locator::effective_replication_map_ptr repair::shard_repair_task_impl::get_erm() noexcept {
+    return erm ? erm : db.local().get_tables_metadata().get_table(table_ids[0]).get_effective_replication_map();
```

Comment:
This returns wrong erm if the caller is asking for erm other than the first table.